neurolib/models/aln/model.py

```diff
@@ -46,61 +77,35 @@ def __init__(
-        if simulateBOLD:
-            self.simulateChunkwise = True  # Override this setting if BOLD is simulated!
-        self.chunkSize = (
-            chunkSize  # Size of integration chunks in chunkwise integration in case of simulateBOLD == True
-        )
-        # Save data from all chunks? Can be very memory demanding if simulations are long or large
-        self.saveAllActivity = saveAllActivity
+        integration = ti.timeIntegration
 
         # load default parameters if none were given
         if params is None:
-            self.params = dp.loadDefaultParams(
-                Cmat=self.Cmat, Dmat=self.Dmat, lookupTableFileName=self.lookupTableFileName, seed=self.seed,
+            params = dp.loadDefaultParams(
+                Cmat=self.Cmat, Dmat=self.Dmat, lookupTableFileName=self.lookupTableFileName, seed=self.seed
             )
-        else:
-            self.params = params
 
-    def run(self):
-        """
-        Runs an aLN mean-field model simulation.
-        """
-
-        if self.simulateChunkwise:
-            t_BOLD, BOLD, return_tuple = cw.chunkwiseTimeIntAndBOLD(
-                self.params, self.chunkSize, self.simulateBOLD, self.saveAllActivity
-            )
-            (rates_exc, rates_inh, t, mufe, mufi, IA, seem, seim, siem, siim, seev, seiv, siev, siiv,) = return_tuple
-            self.t_BOLD = t_BOLD
-            self.BOLD = BOLD
-            self.setOutput("BOLD.t_BOLD", t_BOLD)
-            self.setOutput("BOLD.BOLD", BOLD)
-        else:
-            (
-                rates_exc,
-                rates_inh,
-                t,
-                mufe,
-                mufi,
-                IA,
-                seem,
-                seim,
-                siem,
-                siim,
-                seev,
-                seiv,
-                siev,
-                siiv,
-            ) = ti.timeIntegration(self.params)
+        # Initialize base class Model
+        super().__init__(
+            integration=integration,
+            params=params,
+            state_vars=self.state_vars,
+            init_vars=self.init_vars,
+            output_vars=self.output_vars,
+            input_vars=self.input_vars,
+            default_output=self.defaultOutput,
+            bold=bold,
+            name=self.name,
+            description=self.description,
+        )
 
-        # convert output from kHz to Hz
-        rates_exc = rates_exc * 1000.0
-        rates_inh = rates_inh * 1000.0
+    def getMaxDelay(self):
+        # compute maximum delay of model
+        dt = self.params["dt"]
+        Dmat = dp.computeDelayMatrix(self.params["lengthMat"], self.params["signalV"])
+        Dmat_ndt = np.around(Dmat / dt)  # delay matrix in multiples of dt
+        ndt_de = round(self.params["de"] / dt)
+        ndt_di = round(self.params["di"] / dt)
+        max_global_delay = int(max(np.amax(Dmat_ndt), ndt_de, ndt_di))
+        return max_global_delay
```

Comment:
this is not much different from the superclass' implementation of `getMaxDelay` is it? you just also check `de` and `di` from params, right? so you could run the superclass' function and then just check whether local delays are not larger than `Dmat` max right? what about

```python
def getMaxDelay(self):
    ndt_de = round(self.params["de"] / dt)
    ndt_di = round(self.params["di"] / dt)
    max_dmat_delay = super().getMaxDelay()
    return int(max(max_dmat_delay, ndt_de, ndt_di))
```

Reply:
The parameters `de` and `di` are very specific for the `aln` model, the other models don't have local delays, that's why they can rely on the superclass' `getMaxDelay()`. I didn't want to add another "register your delays" type of functionality, which I why I simply chose to overload the function here. 

Your suggested change is nicer, I will add it! Thanks!